### ISSUE TITLE
airflow_backfill overrides start_date

### ISSUE THREAD

**Issue by criccomini** [apache/airflow#1089]
I ran an `airflow backfill` command as follows:

```
airflow backfill -sd /etc/airflow/dags -s 2014-08-06 -e 2014-08-07 monolith-db-daily
```

My dag (`monolith-db-daily`) had never been run, and had a start_date of datetime.now(). The observed behavior was that the backfill began on 2014-08-02, and continued past 2014-08-07 all the way to present day.

In addition, the DAG was scheduled to run daily at 4am. I observed scheduled execution_dates every day at 4am, for 2014-08-02 through present day, BUT I ALSO observed a second execution on 2014-08-06 and 2014-08-07 at 00:00. I suspect `airflow` backfill is making some assumption about time that's not correct if you schedule daily executions other than 00:00.

**Comment by criccomini** [comment]
My current guess for the continued backfill is that past the 2014-08-07 end date is that once the DAG had its first DAG run at 2014-08-02, it used that at the start_date going forward, and disregarded my DAG's start_date that was defined in Python.

I expected the execution just to back fill those two days, and then pick up on the start_date defined in my Python file for further executions.

**Comment by criccomini** [comment]
I have no explanation as to why the DAG runs started on 2014-08-02. My `-s` was clearly 2014-08-06.

**Comment by criccomini** [comment]
Note: this was on 1.6.2

**Comment by jlowin** [comment]
Are you running a scheduler? If so I think this is the culprit: https://github.com/airbnb/airflow/blob/5ab256780ab4006f4a6fc1d1ef5d0f07d4e934ca/airflow/jobs.py#L417

If a DAG has not been scheduled before, but has been run, then the scheduler tries to start it up from 5 days before the last run. It's a little odd -- I have no idea what the significance of the 5 days is. When I came across it I didn't want to take it out completely because it might have deeper meaning, so I added a catch so that it won't schedule prior to the `start_date`.
